api/server/middleware/middleware_test.go

```diff
@@ -132,6 +149,7 @@ func TestAcceptEncodingHeaderHandler(t *testing.T) {
 	dummyContent := "Test gzip middleware content"
 	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", r.Header.Get("Accept"))
+		w.WriteHeader(http.StatusOK)
```

Comment:
Why do you explicitly call this? Isn't it implicitly called when `Write` is called and no other status code has been set? I think WriteHeader always gets called under the hood anyway so this is fine, but I'm asking to double check that understanding and make sure you didn't add this because leaving it out meant the method didn't get called.